src/modules/QtQuick/Item.js

```diff
@@ -188,9 +188,7 @@ registerQmlType({
   $onDataChanged(newData) {
     for (const i in newData) {
       const child = newData[i];
-      if (child.hasOwnProperty("parent")) {
-        // Seems to be an Item.
-        // TODO: Use real inheritance and ask using instanceof.
+      if (child instanceof QmlWeb.getConstructor("QtQuick", "2.0", "Item")) {
```

Comment:
Could you save this as `QMLItem` at the top of the function (i.e. outside the loop)?